src/GapicClientTrait.php

```diff
@@ -183,6 +183,16 @@ private function buildClientOptions(array $options)
             $options['apiEndpoint'] = $this->pluck('serviceAddress', $options, false);
         }
 
+        // Set scopes if an API endpoint is set to ensure the "audience"
+        // does not conflict with the custom endpoint.
+        if (empty($options['credentialsConfig']['scopes'])
+                && $options['apiEndpoint'] != $defaultOptions['apiEndpoint']
+                && isset($options['defaultScopes'])) {
+            $options['credentialsConfig']['scopes'] = $options['defaultScopes'];
```

Comment:
This ensures when `apiEndpoint` option has been changed, self-signed JWTs are not used. We will be able to remove this once mTLS accepts standard audience strings.